### DISCUSSION TITLE
V2B in Homey

### DISCUSSION THREAD

**Discussion by No-World-Order** [visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA#641] [Show and tell]
I looked at many discussions but could not find anyone using this with Homey. When i received my parts last weekend i started working on it and got everything working now!

Homey works with virtual devices so i added the device and added 3 sections:

- Temperature (set and current)
- Commands (heater and pump on/off)
- Error display

These are being filled by the MQTT flow.
Next to that the heater is disabled when the power prices are the higest of the day. 

It took me some time to get the right commands, so i figured it might be helpfull for those also using Homey 💯 

![1](https://github.com/visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA/assets/139592234/78e87913-2b8d-4128-b91f-9f2daa620b91)
![2](https://github.com/visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA/assets/139592234/9941785f-b943-46d7-afd9-eb269622460b)
![3](https://github.com/visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA/assets/139592234/f492ab0e-d97e-49a1-a591-f674293b2160)
![5](https://github.com/visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA/assets/139592234/1e1085a7-dbe3-4101-a6fd-195de21e5c4d)

**Comment by Stargits** [comment]
Hi,
Thank you for the guide on how to setup this with homey. I've tried with the same solution but I think I get some issues when configuring the MQTT. Could you show your setup of the MQTT connection both in homey and the webinterface of the ESP?

**Reply by visualapproach** [reply]
@Stargits if @No-World-Order is not responding you can post pics on your homey mqtt setup page and we can see if we can figure it out. I know nothing about homey.

**Reply by No-World-Order** [reply]
Thanks for the tag! For MQTT on the homey you need to have 3 components:

MQTT Broker
MQTT Client
MQTT Hub

For the MQTT Broker you can use the default ports: 1883, tick "also allow unsecure connections".
At the user section add a user with your preferred user and password.

As I've removed my SPA during the winter i'm not sure how that config looked like in detail, but generally you add the IP of your homey there and fill in the ports + user and password. If you have any screens i can talk you through.

**Reply by Stargits** [reply]
Thanks again for a good explanation!
It seems to be working now since I receive messages from the ESP in MQTT Client.
Now I'm just going to install it in the spa itself and test it all together.

![LayZSpa MQTT Config](https://github.com/visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA/assets/157060948/d23d0673-35d9-4435-99bd-334a97d5314d)
![Homey_MQTTClient_Log](https://github.com/visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA/assets/157060948/e7e2129f-5ac8-4e56-a2c3-aaaa10bc8390)
![Homey_MQTTBroker_Config](https://github.com/visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA/assets/157060948/4b061337-b0ea-43c5-8f70-954c575b2087)

**Comment by MaciverVFR** [comment]
@Stargits and @No-World-Order 
Hope you guys may help.
Got it all set up and i´m able to set Target temp, switch on and off pump, heat and bubbles, but i can´t get the current temp to show in the Homey.
Using the app "Device Capabilities" to generate the virtual parts.
When i try the flow, it fails.
What am i missing?
If i do changes to the buttons on the pump or in the V2B webb the Homey doesn´t change my "power" buttons status. Any suggestions on how to make tham change ?
I have set the telemetry interval to 60 sek.
![Skärmbild 2024-05-05 163209](https://github.com/visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA/assets/168960940/eff7afd5-5556-402b-808a-a5e0b7d62f2f)
![Skärmbild 2024-05-05 184323](https://github.com/visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA/assets/168960940/e73cb76a-2b45-48a3-acf5-8b09e1ada331)

**Reply by Stargits** [reply]
Hi @MaciverVFR ,
I would double check the setting of your virtual device capabilities. As in my setup below I have one number field for the actual temperature and one for the target temperature. Make sure that you update the correct one in the flow.

Then regarding next issue with updating your buttons I had the same issue. Try swap the order of calling the functions in the flow.
"Trigger when a message is recieved" from MQTT should be in the top of the flow to make sure you receive the latest data before you execute any commands. Then it will work.

![image](https://github.com/visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA/assets/157060948/9d4ed96c-7c00-4f7b-bae3-3abe03b80ff1)

![image](https://github.com/visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA/assets/157060948/97d98694-3925-4384-b68a-ce1d06244c19)

**Comment by MrFortunaBeach** [comment]
I have tried this but cannot find the correct Homey Virtual Device App. Plz help.

**Reply by MaciverVFR** [reply]
I´ve got "Device Capabilities" and "Virtual Devices" installed and i don´t remember wich one i used, Belive i tried both and didn´t unistall the one i didn´t use.

**Reply by Stargits** [reply]
Im using ”device capabilities” for this.

**Reply by MrFortunaBeach** [reply]
Thanks a lot. Now I need to figure out things :-)
/BR
Thomas

**Reply by MrFortunaBeach** [reply]
My god, Why cant I figure this out :-(

**Reply by Stargits** [reply]
Do you receive any values from the ESP on MQTT or where do you get stuck?

**Reply by MrFortunaBeach** [reply]
Hi sorry for the delay.
I have my MQTT Server up n Running in Homey and the LayZSpa listnen to commands.
I Can control On/Off by using Virtual Devices but I want to Do it moore fancy with Device Capabilities. So I need to install
![MQTT](https://github.com/user-attachments/assets/c601c8c3-5a1e-46ce-8da8-dde17fc556fb)
![MQTT2](https://github.com/user-attachments/assets/df6b89e8-26a3-40d5-b27c-462dbb69d3d2)
I really get stuck configuring the Device Capabilities and Configure the flow for the control and temperature reading(I´n not a programmer)

BR
Thomas

**Reply by Stargits** [reply]
I see, it’s a bit tricky but if you take it step by step it shouldnt be a problem. Start with adding a setpoint value and an actual value in device capabilities as previous screenshots. It will look something lile this:
![IMG_2633](https://github.com/user-attachments/assets/484717a9-b0d5-4eba-b675-3db5c7330e2c)
Next step is to update the actual value when a new message is received on mqtt using a flow as previous screenshots.
After that you will need to send a message if the setpoint is changed.
Hope it makes sense.
